cmd/descheduler/app/options/options.go

```diff
@@ -0,0 +1,67 @@
+package options
+
+import (
+	"time"
+
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	componentbaseconfig "k8s.io/component-base/config"
+
+	"github.com/karmada-io/karmada/pkg/util"
+)
+
+const (
+	defaultBindAddress            = "0.0.0.0"
+	defaultPort                   = 10351
+	defaultEstimatorPort          = 10352
+	defaultDeschedulingInterval   = 2 * time.Minute
+	defaultUnschedulableThreshold = 5 * time.Minute
+)
+
+// Options contains everything necessary to create and run scheduler-estimator.
+type Options struct {
+	LeaderElection componentbaseconfig.LeaderElectionConfiguration
+	KubeConfig     string
+	Master         string
+	// BindAddress is the IP address on which to listen for the --secure-port port.
+	BindAddress string
+	// SecurePort is the port that the server serves at.
+	SecurePort int
+
+	KubeAPIQPS float32
+	// KubeAPIBurst is the burst to allow while talking with karmada-apiserver.
+	KubeAPIBurst int
+
+	// SchedulerEstimatorTimeout specifies the timeout period of calling the accurate scheduler estimator service.
+	SchedulerEstimatorTimeout metav1.Duration
+	// SchedulerEstimatorPort is the port that the accurate scheduler estimator server serves at.
+	SchedulerEstimatorPort int
+	// DeschedulingInterval specifies time interval for descheduler to run.
+	DeschedulingInterval metav1.Duration
+	// UnschedulableThreshold specifies the period of pod unschedulable condition.
+	UnschedulableThreshold metav1.Duration
+}
+
+// NewOptions builds an empty options.
+func NewOptions() *Options {
+	return &Options{}
+}
+
+// AddFlags adds flags of estimator to the specified FlagSet
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	if o == nil {
+		return
+	}
+	fs.BoolVar(&o.LeaderElection.LeaderElect, "leader-elect", true, "Enable leader election, which must be true when running multi instances.")
+	fs.StringVar(&o.LeaderElection.ResourceNamespace, "leader-elect-resource-namespace", util.NamespaceKarmadaSystem, "The namespace of resource object that is used for locking during leader election.")
+	fs.StringVar(&o.KubeConfig, "kubeconfig", o.KubeConfig, "Path to a KubeConfig. Only required if out-of-cluster.")
+	fs.StringVar(&o.Master, "master", o.Master, "The address of the Kubernetes API server. Overrides any value in KubeConfig. Only required if out-of-cluster.")
```

Comment:
Which flag is for karmada kubeconfig? 

Remind me why rescheduled need to access host cluster?

Reply:
This flag is karmada kubeconfig. The same as https://github.com/karmada-io/karmada/blob/master/cmd/scheduler/app/options/options.go#L30